workloads/templates/vm-creation.md

```diff
@@ -77,29 +103,118 @@ $ oc process -f cluster/vmi-template-fedora.yaml\
 
 ``` bash
-$ oc patch virtualmachine testvmi --type merge -p '{"spec":{"running":true}}'
-virtualmachine "testvmi" patched
+$ oc patch virtualmachine testvm --type merge -p '{"spec":{"running":true}}'
+virtualmachine.kubevirt.io/testvm patched
 ```
 
+Do not forget about virtctl tool. Using it in the real cases instead of using kubernetes API can be more convinient. Example: 
+
+```bash
+$ virtctl start testvm
+VM testvm was scheduled to start
+```
+
+As soon as VM starts, kubernates creates new type of object - VirtualMachineInstance. It has similar name to VirtualMachine. Example (not full output, it's too big):
+
+```bash
+$ kubectl describe vm testvm
+name:         testvm
+Namespace:    myproject
+Labels:       kubevirt-vm=vm-testvm
+              kubevirt.io/os=fedora27
+Annotations:  <none>
+API Version:  kubevirt.io/v1alpha2
+Kind:         VirtualMachine
+.......
+```
+
+## Cloud-init script and parameters
+
+Kubevirt VM templates, just like kubevirt VM/VMI yaml configs, supports [cloud-init scripts](https://cloudinit.readthedocs.io/en/latest/)
+
+
+## Using registry images
+
+Kubevirt VM templates, just like kubevirt VM/VMI yaml configs, supports creating VM's disks from registry. RegistryDisk is a special type volume which supports downloading images from user-defined registry server.
+
+## **Hack** - use pre-downloaded image
+
+Kubevirt VM templates, just like kubevirt VM/VMI yaml configs, can use pre-downloaded VM image, which can be a useful feature especially in the debug/development/testing cases. No special parameters required in the VM template or VM/VMI yaml config. The main idea is to create Kubernetes PersistentVolume and PersistentVolumeClaim corresponding to existing image in the file system. Example:
+
+```yaml
+---
+kind: PersistentVolume
+apiVersion: v1
+metadata:
+  name: mypv
+  labels:
+    type: local
+spec:
+  storageClassName: manual
+  capacity:
+    storage: 10G
+  accessModes:
+    - ReadWriteOnce
+  hostPath:
+    path: "/mnt/sda1/images/testvm"
+---
+kind: PersistentVolumeClaim
+apiVersion: v1
+metadata:
+  name: mypvc
+spec:
+  storageClassName: manual
+  accessModes:
+    - ReadWriteOnce
+  resources:
+    requests:
+      storage: 10G
+
+```
+
+If you create this PV/PVC, then you have to put VM image in the file path
+
+```bash
+/mnt/sda1/images/testvm/disk.img
+```
+
+Avaible in the each OpenShift/Kubevirt compute nodes.
+
+## WebUI
+Kubevirt project has [the official UI](https://github.com/kubevirt/web-ui). This UI supports creation VM using templates and templates features - flavors and workload profiles.
+To create VM from template, choose WorkLoads in the left panel >> press to the "Create Virtual Machine" blue button >> choose "Create from Wizzard". Next, you have to see "Create Virtual Machine" window
+
+## common-templates
```

Comment:
As I mentioned above, this should be much more visible I think. The user should prefer using the pre-created templates instead of writing his own.

Reply:
>Maybe move this up in the document?
Agree

Reply:
Moved

Reply:
Not really, you moved the UI section, not the template section. What I would do is moving the common-templates section to the top and then below it writing something like: "You can also create templates by hand and here is an example of how it could look like" and then I would use the existing Example template section